### PR TITLE
adding center parameters for AllSkyAxes

### PR DESCRIPTION
**Pull request by swyatt7**
Adds the ability to center a mollweide or aitoff projection on given coordinates. 

The `center` parameter follows the same behavior as the zoom and globe projections. It is defaulted to `"180d 0d"` upon init in the `AllSkyAxes` class, and cast into an `astropy.coordinates.SkyCoord`, where the ra and dec are set to the header `"CRVAL1"` and `"CRVAL2"`, respectively.

### CODE REVIEW

**Review by lpsinger — CHANGES_REQUESTED** [review]
- [ ] `180d 0d` is the correct default value for `astro` projections but not `geo` projections — hence the original setting of `'CRVAL1': self._crval1`. Please fix.
- [ ] Please add image comparison tests to `ligo.skymap.plot.tests.test_plot`.

File: ligo/skymap/plot/tests/test_plot.py [hunk 1]
```diff
@@ -151,3 +151,14 @@ def test_reticle():
...
+
+@pytest.mark.parametrize('proj', ['aitoff', 'mollweide'])
+@pytest.mark.parametrize('cent', ['197.45d -23.38d', None])
+@pytest.mark.mpl_image_compare(remove_text=True, tolerance=1.5)
+def test_center_projections(rcparams, proj, cent):
+    fig = plt.figure(figsize=(4, 4))
+    ax = fig.add_axes(111, projection=f'astro {proj}',
+                      center=cent)
+    ax.grid()
+    return fig
```
Comment (lpsinger) [inline ligo/skymap/plot/tests/test_plot.py:164]: Add missing newline to end of file.

File: ligo/skymap/plot/allsky.py [hunk 2]
```diff
@@ -316,6 +316,7 @@ def __init__(self, *args, header, obstime=None, **kwargs):
         h['CDELT1'] /= scale1
         h['CDELT2'] /= scale2
         if obstime is not None:
+            h['MJD-OBS'] = Time(obstime).utc.mjd
```
Comment (lpsinger) [inline ligo/skymap/plot/allsky.py:319]: Why is it necessary to set MJD-OBS in the header?

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Comment by lpsinger** [comment]
Also, please add a change log entry!

File: ligo/skymap/plot/allsky.py (hunk 2 above)
Comment (swyatt7) [reply ligo/skymap/plot/allsky.py:319]: It suppresses this warning from astropy.wcs:
```
ligo/skymap/plot/tests/test_plot.py: 13 warnings
  /Users/crisp/anaconda3/envs/ligosky/lib/python3.11/site-packages/astropy/wcs/wcs.py:819: FITSFixedWarning: 'datfix' made the change 'Set MJD-OBS to 57982.528524 from DATE-OBS'.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

I can revert it back if you see necessary.

File: ligo/skymap/plot/tests/test_plot.py (hunk 1 above)
Comment (swyatt7) [reply ligo/skymap/plot/tests/test_plot.py]: in recent push

File: CHANGES.rst [hunk 3]
```diff
@@ -5,7 +5,7 @@ Changelog
 1.1.3 (unreleased)
 ==================
 
-- No changes yet.
+- Added options to center `mollweide` and `aitoff` projections.
```
Comment (lpsinger) [inline CHANGES.rst:8]: This is reStructuredText, not Markdown:

```suggestion
- Added options to center ``mollweide`` and ``aitoff`` projections.
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Comment by lpsinger** [comment]
This looks great! This is a GitHub mirror of https://git.ligo.org/lscsoft/ligo.skymap/ and the real CI/CD pipeline runs in GitLab, so I pushed this to a branch and opened it as a GitLab MR here: https://git.ligo.org/lscsoft/ligo.skymap/-/merge_requests/343

**Review by lpsinger — CHANGES_REQUESTED** [review]
Please fix linter errors from https://git.ligo.org/leo-singer/ligo.skymap/-/jobs/3061659.

**Review by lpsinger — CHANGES_REQUESTED** [review]
Please update this documentation: https://github.com/lpsinger/ligo.skymap/blob/68fca3ddf1e54788fe5ab19bffabe58e987bb2f1/ligo/skymap/plot/allsky.py#L44-L46

**Review by lpsinger — CHANGES_REQUESTED** [review]
Please fix lint errors: https://git.ligo.org/leo-singer/ligo.skymap/-/jobs/3063950

**Pull request closed without merging**